### DISCUSSION TITLE
custom control instead of pin

### DISCUSSION THREAD

**Discussion by golosalex** [Mapsui/Mapsui#1592] [General]
can i somehow use my custom user control instead of pin? i want to generate custom visual for it and make some interactive, than place my custom control in right place on the map.

**Comment by inforithmics** [comment]
I think you should create a widget for that, see implemented widgets in mapsui.

**Comment by pauldendulk** [comment]
With a widget you can draw your own control with skia. Is that what you are looking for? Or would were you thinking of a Xamarin custom control? (or whatever platform you use).

**Comment by golosalex** [comment]
I whant to use costom WPF control with binding to my ViewModel data. i whant to manipulate whith size and shapes when my VM chanched and binding comands to controls on map.

Mapsui and using computer's map is pretty new for me.
Can a widget handle any event?
